GroceryList/app/src/main/java/com/mtruehle/grocerylist/MainActivity.java

```diff
@@ -0,0 +1,152 @@
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        itemsToBuy = new ArrayList<String>();
+        itemsToBuy.add("Lucky Charms"); // they're magically delicious.
+        itemsAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, itemsToBuy);
+        ListView itemListView = (ListView) findViewById(R.id.groceryList);
+        itemListView.setAdapter(itemsAdapter);
+        itemListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView parent, View view, int position, long id) {
+                handleListClick(view, position); // separating the dialog and handling for clicking, mainly for ease of reading and debugging.
+            }
+        });
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.menu_main, menu);
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.delete_all:
+                AlertDialog.Builder areYouSure = new AlertDialog.Builder(this);
+                areYouSure.setTitle(R.string.are_you_sure);
+                areYouSure.setPositiveButton(R.string.yes, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        itemsToBuy.clear();
+                        itemsAdapter.notifyDataSetChanged();
+                        Toast.makeText(MainActivity.this, R.string.cleared, Toast.LENGTH_SHORT).show();
+                    }
+                });
+                areYouSure.setNegativeButton(R.string.cancel_button, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        // toast here? or is that too much feedback?
+                    }
+                });
+                AlertDialog aD = areYouSure.create();
+                aD.show();
+                return true;
+            default:
+                return true;
+        }
+    }
+
+
+
+    public void handleListClick(View view, int position) {
+        // When an item on a list is clicked, its position is passed here and an AlertDialog is used to ask whether it's being edited, deleted, etc.
+        AlertDialog.Builder editOrDelete = new AlertDialog.Builder(this);
+        final int pos = position;
+        editOrDelete.setTitle(R.string.edit_or_delete);
+        editOrDelete.setMessage(itemsToBuy.get(position));
+        editOrDelete.setPositiveButton(R.string.delete_button, new DialogInterface.OnClickListener() {
```

Comment:
Again, it will keep your code nice and clean and separated if you define these in separate files.